src/plugins/matrixops/hsplit_operation.cpp

```diff
@@ -0,0 +1,385 @@
+              hpx::bad_parameter, "hsplit_operation::eval",
+              generate_error_message(
+                  "the hsplit_operation primitive can not split "
+                  "matrices/vectors into more blocks than there "
+                  "are columns"));
+        }
+        if (num_cols % num_blocks != 0) {
+          HPX_THROW_EXCEPTION(
+              hpx::bad_parameter, "hsplit_operation::eval",
+              generate_error_message(
+                  "the hsplit_operation primitive can not split "
+                  "matrices/vectors unevenly"));
+        }
+        std::size_t block_size = num_cols / num_blocks;
+
+        std::size_t index(num_rows);
+        std::size_t tmp_block_num(num_blocks);
+
+        for (int i = 0; i < num_cols; i += block_size) {
+          std::pair<std::size_t, std::size_t> range(i, i + block_size);
+          ranges.push_back(range);
+        }
+        for (int i = 0; i < ranges.size(); i++) {
+          std::cout << ranges[i].first << " " << ranges[i].second << std::endl;
+        }
+      } else // Index-based tiling
+      {
+        auto &&indices = extract_node_data<T>(std::move(args[1]));
+        auto v = indices.vector();
+
+        if (v.size() > 0) {
+          for (int i = 0; i < v.size(); i++) {
+            if (static_cast<std::size_t>(v[i]) > num_cols)
+              v[i] = num_cols;
+          }
+
+          // Include from 0 to first index
+          ranges.push_back(std::make_pair<std::size_t, std::size_t>(
+              std::size_t(0), std::size_t(v[0])));
+
+          // Make a list of pairs from the passed index list
+          for (int i = 0; i < v.size() - 1; i++) {
+            ranges.push_back(std::make_pair<std::size_t, std::size_t>(
+                std::size_t(v[i]), std::size_t(v[i + 1])));
+          }
+
+          // Include from last index to last row
+          ranges.push_back(std::make_pair<std::size_t, std::size_t>(
+              std::size_t(v[v.size() - 1]), std::size_t(num_cols)));
+        } else {
+          HPX_THROW_EXCEPTION(hpx::bad_parameter, "hsplit_operation::eval",
+                              generate_error_message(
+                                  "the hsplit_operation primitive requires all "
+                                  "index vectors to be of size at least 1"));
+        }
+      }
+
+      std::int64_t num_blocks = ranges.size();
+
+      primitive_arguments_type result;
+      result.reserve(num_blocks);
+
+      auto &&input_data = extract_node_data<T>(std::move(args[0]));
+      auto m = input_data.matrix();
+
+      for (int i = 0; i < num_blocks; i++) {
+        if (ranges[i].second > ranges[i].first) {
+          std::size_t diff = ranges[i].second - ranges[i].first;
+          blaze::DynamicMatrix<T> block(num_rows, diff);
+          for (int j = 0; j < num_rows; j++) {
+            for (int k = 0; k < diff; k++) {
+              block(j, k) = m(j, ranges[i].first + k);
+            }
+          }
```

Comment:
I think it should be possibl;e to use blaze views (`blaze::rows`/`blaze::columns`) for this to simplify the code.